### ISSUE TITLE
"One CLI for webpack must be installed" after minor update from 4.44.2 to 4.45.0/4.46.0

### ISSUE THREAD

**Issue by caugner** [webpack/webpack#12655]
# Bug report

**What is the current behavior?**

After upgrading webpack from 4.44.2 to 4.45.0 (or 4.46.0), webpack asks to install `webpack-cli`:

```console
$ yarn prod
yarn run v1.22.5
$ npm run production
npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1612990005632-0.9294932316423565/node but npm is using /home/dev/.asdf/installs/nodejs/14.15.4/bin/node itse
lf. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> @0.31.1 production /home/dev/git/acme
> cross-env NODE_ENV=production node_modules/webpack/bin/webpack.js --no-progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js



 DONE  Compiled successfully in 42383ms          

$ yarn list webpak
yarn list v1.22.5
warning Filtering by arguments is deprecated. Please use the pattern option instead.
├─ @vue/cli-plugin-babel@4.5.11
│  └─ webpack@4.46.0
├─ @vue/cli-plugin-eslint@4.5.11
│  └─ webpack@4.46.0
├─ @vue/cli-service@4.5.11
│  └─ webpack@4.46.0
└─ webpack@4.44.2
Done in 0.89s.

$ yarn upgrade webpack
...

$ yarn list webpack
yarn list v1.22.5
warning Filtering by arguments is deprecated. Please use the pattern option instead.
├─ laravel-mix@5.0.9
│  └─ webpack@4.44.2
└─ webpack@4.46.0
Done in 0.89s.

$ yarn prod
yarn run v1.22.5
$ npm run production
npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1612989379362-0.18985632006525122/node but npm is using /home/dev/.asdf/installs/nodejs/14.15.4/bin/node its
elf. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> @0.31.1 production /home/dev/git/acme
> cross-env NODE_ENV=production node_modules/webpack/bin/webpack.js --no-progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js

One CLI for webpack must be installed. These are recommended choices, delivered as separate packages:
 - webpack-cli (https://github.com/webpack/webpack-cli)
   The original webpack full-featured CLI.
We will use "yarn" to install the CLI via "yarn add -D".
Do you want to install 'webpack-cli' (yes/no):
```

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

The minor update should not require the installation of a new dependency.

**Other relevant information:**
webpack version: 4.45.0 / 4.46.0
Node.js version: 14.15.4
Operating System: Ubuntu 20.04.2 LTS via WSL2 (Windows 10)
Additional tools:

**Comment by 368573607** [comment]
Core and cli are separate.

So you must install `webpack-cli`:

```
$ npm i -D webpack-cli
```

You can also install globally. Just add a `-g`.

**Comment by caugner** [comment]
@368573607 Why is this (breaking) change not mentioned in the 4.45.0 release notes? Also, I'm a bit surprised about this new requirement, because the changes between 4.44.2 and 4.45.0 doesn't include anything like that.

**Comment by sokra** [comment]
This is not new. It was changed from v3 to v4. You did need webpack-cli with 4.44.2 too, maybe some of your dependencies installed it and it is now placed somewhere different in your node_modules.

**Comment by alexander-akait** [comment]
Answer above, you need to install `webpack-cli`, it was changed from v3 to v4

**Comment by caugner** [comment]
> Answer above, you need to install `webpack-cli`, it was changed from v3 to v4

 @alexander-akait If I `yarn add -D webpack-cli`, I'm getting `Error: Cannot find module 'webpack-cli/bin/config-yargs'` instead:

```
hot_1       | internal/modules/cjs/loader.js:883
hot_1       |   throw err;
hot_1       |   ^
hot_1       |
hot_1       | Error: Cannot find module 'webpack-cli/bin/config-yargs'
hot_1       | Require stack:
hot_1       | - /var/www/html/node_modules/webpack-dev-server/bin/webpack-dev-server.js
hot_1       |     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
hot_1       |     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
hot_1       |     at Module.require (internal/modules/cjs/loader.js:952:19)
hot_1       |     at require (internal/modules/cjs/helpers.js:88:18)
hot_1       |     at Object.<anonymous> (/var/www/html/node_modules/webpack-dev-server/bin/webpack-dev-server.js:65:1)
hot_1       |     at Module._compile (internal/modules/cjs/loader.js:1063:30)
hot_1       |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
hot_1       |     at Module.load (internal/modules/cjs/loader.js:928:32)
hot_1       |     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
hot_1       |     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12) {
hot_1       |   code: 'MODULE_NOT_FOUND',
hot_1       |   requireStack: [
hot_1       |     '/var/www/html/node_modules/webpack-dev-server/bin/webpack-dev-server.js'
hot_1       |   ]
hot_1       | }
```

**Comment by alexander-akait** [comment]
Please use `webpack serve` to run webpack-dev-server, don't use `webpack-dev-server` directly

**Comment by caugner** [comment]
Thank you so much, that was it.

**Comment by caugner** [comment]
For the record: Even after applying the fix, the update from `4.44.2` to `4.46.0` still broke the build for me:

* With `4.44.2`, our webpack alias (`"@": require("path").join(__dirname, "src")`) worked as expected,
* With `4.46.0` (after a simple `yarn upgrade webpack`), webpack complained about missing dependencies:

```
$ cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js serve --hot --config=node_modules/laravel-mix/setup/webpack.config.js

These dependencies were not found:

* @/App in ./src/main.js
* @/views/HomeView in ./src/router/routes.js
* ...
```

**Reason**: `vue-loader` was missing from the `dependencies`.

**Solution**: `yarn add -D vue-loader` resolved the issue for good.
